### PR TITLE
IGNITE-16665 Introduced integer table id.

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-16665

### CODE REVIEW

**Review by SammyVimes — APPROVED** [review]
looks good to me

File: modules/api/src/main/java/org/apache/ignite/configuration/schemas/table/TableConfigurationSchema.java [hunk 1]
```diff
@@ -35,6 +36,12 @@ public class TableConfigurationSchema {
     @InjectedName
     public String name;
 
+    /** Integer table id. */
+    @Immutable
+    @Range(min = 1, max = Integer.MAX_VALUE)
+    @Value(hasDefault = true)
+    public int intId = 1;
```
Comment (tkalkirill) [inline modules/api/src/main/java/org/apache/ignite/configuration/schemas/table/TableConfigurationSchema.java:43]: Why such a name? seems to be more correct just **id**.
Why such a range?
Can we use UUID as an identifier?

File: modules/api/src/main/java/org/apache/ignite/configuration/schemas/table/TablesConfigurationSchema.java [hunk 2]
```diff
@@ -29,6 +29,10 @@
 @SuppressWarnings("PMD.UnusedPrivateField")
 @ConfigurationRoot(rootName = "table", type = ConfigurationType.DISTRIBUTED)
 public class TablesConfigurationSchema {
+    /** Global integer id counter. */
+    @Value(hasDefault = true)
+    public int globalIdCounter = 0;
```
Comment (tkalkirill) [inline modules/api/src/main/java/org/apache/ignite/configuration/schemas/table/TablesConfigurationSchema.java:34]: Is this the global table identifier? why is it not clearly named? what happens if it overflows?

File: modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStoreManager.java [hunk 3]
```diff
@@ -168,30 +168,30 @@ public long allocatePage(int grpId, int partId, byte flags) throws IgniteInterna
     /**
      * Initializing the file page stores for a group.
      *
-     * @param grpName Group name.
-     * @param grpId Group ID.
+     * @param tableName Table name.
+     * @param groupId Integer table id.
```
Comment (tkalkirill) [inline modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/store/FilePageStoreManager.java:172]: **tableId** ?

File: modules/api/src/main/java/org/apache/ignite/configuration/schemas/table/TableConfigurationSchema.java (hunk 1 above)
Comment (ibessonov) [reply modules/api/src/main/java/org/apache/ignite/configuration/schemas/table/TableConfigurationSchema.java:43]: UUID is already a main "globally unique" identifier. This one - I'm not sure about the name, honestly. Field named "id" already exists.

File: modules/api/src/main/java/org/apache/ignite/configuration/schemas/table/TablesConfigurationSchema.java (hunk 2 above)
Comment (ibessonov) [reply modules/api/src/main/java/org/apache/ignite/configuration/schemas/table/TablesConfigurationSchema.java:34]: I'l expand the comment

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**